subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/ConventionExtensions.kt

```diff
@@ -36,18 +36,50 @@ inline fun <reified T : Any> Convention.getPluginByName(name: String): T =
     } ?: throw IllegalStateException("A convention named '$name' could not be found.")
 
 
-inline fun <reified T : Any> Convention.getPlugin() =
+/**
+ * Locates the plugin convention object with the given type.
+ *
+ * @param T the convention plugin type
+ * @return the convention plugin
+ * @throws [IllegalStateException] when there is no such object contained in this convention, or when there are multiple such objects
+ * @see [Convention.getPlugin]
+ */
+inline fun <reified T : Any> Convention.getPlugin(): T =
     getPlugin(T::class)
 
 
+/**
+ * Locates the plugin convention object with the given type.
+ *
+ * @param conventionType the convention plugin type
+ * @return the convention plugin
+ * @throws [IllegalStateException] when there is no such object contained in this convention, or when there are multiple such objects
```

Comment:
Are you sure this is `IllegalStateException`? Sounds like it should be an `IllegalArgumentException`.

Reply:
Yes, it is, see https://github.com/gradle/gradle/blob/master/subprojects/core-api/src/main/java/org/gradle/api/plugins/Convention.java#L46

Reply:
Strange, not the exception I would have expected.